### PR TITLE
Liberation Station work-around

### PR DESCRIPTION
**Pull request by MMMiracles**
work-around for not being able to directly place vending machines. still need to spawn in new refill units but other then that it works.

:steam_locomotive: :us::us::us: cant stop the patriot train choo choo

### CODE REVIEW

**Comment by MMMiracles** [comment]
also if theres any chance of a quick merge ( i have no idea how things like this work around here ) that'd be great.

**Comment by WJohn** [comment]
Quickmerges sadly only happen for high and critical priority fixes. Everything else gets the arbitrary 24 hour review period where conflicts get plenty of time to ruin your day.

**Comment by PKPenguin321** [comment]
24 hour period isn't that all bad, it gives badmins a chance to see the PR and be notified of the freedom they are now able to hand out

**Comment by Fayrik** [comment]
That's call, but as a professional badmin I discovered today that to get a spawned vending machine to work all you need to do is call the initialize() proc.
I'm honestly not sure why that proc even exists, but someone who is more familiar with the code probably will. It might be a better idea to just make initialize part of the creation processes.

File: code/game/objects/items/weapons/vending_items.dm [hunk 1]
```diff
@@ -34,6 +34,11 @@
 	icon_state = "refill_booze"
 	charges = 46//of 138
 
+/obj/item/weapon/vending_refill/liberationstation
+	machine_name = "Liberation Station"
```
Comment (Iamgoofball) [inline code/game/objects/items/weapons/vending_items.dm]: should be "Freedom Resupply Air Drop"

Comment (francinum) [inline code/game/objects/items/weapons/vending_items.dm]: :+1:

**Comment by phil235** [comment]
Doesn't work, spawned liberation vending machine are still empty. If you deconstruct and reconstruct it then the list of products appear but each product has zero items left.

Also, that fix would mean someone screwdriving a vending machine circuitboard can make it select liberation type, which is bad.

**Comment by MMMiracles** [comment]
Like I said, you need to spawn in new refill units because I have literally no idea why they come out empty.

Also actually building the vendor requires 3 of the refill units, which are admin-spawn only, so at the worst they'd have an incomplete machine frame, unless thats also bad.

**Comment by Iamgoofball** [comment]
Why dont we just add initialize() to vending machine new calls

**Comment by phil235** [comment]
I made PR ( #8366) to fix the underlying issue.

**Comment by Jordie0608** [comment]
Since #8366 was merged the work around code can be removed from this.

File: code/game/objects/items/weapons/vending_items.dm [hunk 2]
```diff
@@ -34,6 +34,11 @@
 	icon_state = "refill_booze"
 	charges = 46//of 138
 
+/obj/item/weapon/vending_refill/liberationstation
+	machine_name = "Freedom Supply Drop"
```
Comment (GunHog) [inline code/game/objects/items/weapons/vending_items.dm]: I loooooove this name.

**Comment by Jordie0608** [comment]
Is this going to be updated?

**Comment by WJohn** [comment]
This is kind of obsolete now that the liberty vendors work and can be spawned.

**Comment by Jordie0608** [comment]
They lack a vendor refill pack, which was one of the things added here.

**Comment by MMMiracles** [comment]
Been busy with shit lately,  I'll probably look into fixing it real quick later today.

**Comment by phil235** [comment]
As I said, with this PR you'd be able to screwdrive a vending machine circuitboard and select liberation type. Sure you wouldn't be able to construct it but it still kind of breaks the immersion.

**Comment by phil235** [comment]
Closing this, it wasn't needed.

**Pull request closed without merging**